src/utils/__tests__/internal.ts

```diff
@@ -660,3 +671,49 @@ describe('isModuleOrTagMatch', () => {
     expect(result).toEqual(false);
   });
 });
+
+describe('assertExpectedChainVersion', () => {
+  let client: MockWebSocket;
+  beforeAll(() => {
+    dsMockUtils.initMocks();
+  });
+
+  beforeEach(() => {
+    client = getWebSocketInstance();
+  });
+
+  afterEach(() => {
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    dsMockUtils.cleanup();
+  });
+
+  it('should resolve if it receives expected chain version', () => {
+    const signal = assertExpectedChainVersion('ws://example.com');
+    client.onopen();
+
+    return expect(signal).resolves.not.toThrow();
+  });
+
+  it('should throw an error given an unexpected version', () => {
+    const signal = assertExpectedChainVersion('ws://example.com');
+    client.sendVersion('3.0.0');
+    const expectedError = new PolymeshError({
+      code: ErrorCode.FatalError,
+      message: 'Unsupported Polymesh version. Please upgrade the SDK',
+    });
+    return expect(signal).rejects.toThrowError(expectedError);
+  });
+
+  it('should throw an error if the version does not match', () => {
```

Comment:
The description doesn't match what is being tested

Reply:
My bad, you are right, will update